Decisions.TruBot/Steps/BotSteps.cs

```diff
@@ -0,0 +1,268 @@
+                
+                string result = TruBotRest.TruBotPost(url, auth, content);
+                TruBotResponse response = TruBotResponse.JsonDeserialize(result);
+
+                return response;
+            }
+            catch (Exception ex)
+            {
+                throw new BusinessRuleException("The request to TruBot was unsuccessful.", ex);
+            }
+        }
+
+        public JobStatusResponse GetJobStatusByJobId(int jobId,
+            [IgnoreMappingDefault, PropertyClassification(0, "Override Base URL", "Settings")] string? overrideBaseUrl)
+        {
+            if (jobId == null)
+            {
+                throw new BusinessRuleException("jobId cannot be null.");
+            }
+
+            string baseUrl = ModuleSettingsAccessor<TruBotSettings>.GetSettings().GetBotUrl(overrideBaseUrl);
+            
+            TruBotAuthentication auth = new TruBotAuthentication
+            {
+                Token = ModuleSettingsAccessor<TruBotSettings>.GetSettings().Token,
+                Sid = ModuleSettingsAccessor<TruBotSettings>.GetSettings().Sid
+            };
+
+            try
+            {
+                JobIdRequest inputs = new JobIdRequest();
+                inputs.JobId = jobId;
+                
+                JsonContent content = JsonContent.Create(inputs);
+                
+                string result = TruBotRest.TruBotPost($"{baseUrl}/GetJobStatus", auth, content);
+                
+                return JobStatusResponse.JsonDeserialize(result);
+            }
+            catch (Exception ex)
+            {
+                throw new BusinessRuleException("The request to TruBot was unsuccessful.", ex);
+            }
+        }
+        
+        public JobStatusResponse GetJobStatusByJobExecutionId(string jobExecutionId,
+            [IgnoreMappingDefault, PropertyClassification(0, "Override Base URL", "Settings")] string? overrideBaseUrl)
+        {
+            if (string.IsNullOrEmpty(jobExecutionId))
+            {
+                throw new BusinessRuleException("jobExecutionId cannot be null or empty.");
+            }
+
+            string baseUrl = ModuleSettingsAccessor<TruBotSettings>.GetSettings().GetBotUrl(overrideBaseUrl);
+            
+            TruBotAuthentication auth = new TruBotAuthentication
+            {
+                Token = ModuleSettingsAccessor<TruBotSettings>.GetSettings().Token,
+                Sid = ModuleSettingsAccessor<TruBotSettings>.GetSettings().Sid
+            };
+
+            try
+            {
+                JobExecutionIdRequest inputs = new JobExecutionIdRequest();
+                inputs.JobExecutionId = jobExecutionId;
+
+                JsonContent content = JsonContent.Create(inputs);
+                
+                string result = TruBotRest.TruBotPost($"{baseUrl}/GetJobStatus", auth, content);
+                
+                return JobStatusResponse.JsonDeserialize(result);
```

Comment:
If something goes wrong in the Post message, it will return a business rule exception with a related status code, like 401 (Unauthorized) for instance